src/stormpath.social-login.js

```diff
@@ -0,0 +1,143 @@
+'use strict';
+/**
+ * @ngdoc overview
+ *
+ * @name  stormpath.socialLoginService
+ *
+ * @description
+ *
+ * This module provides the {@link stormpath.socialLoginService.$socialLogin $socialLogin} service.
+ *
+ * Currently, this provider does not have any configuration methods.
+ */
+
+/**
+ * @ngdoc object
+ *
+ * @name stormpath.socialLoginService.$socialLoginProvider
+ *
+ * @description
+ *
+ * Provides the {@link stormpath.socialLoginService.$socialLogin $socialLogin} service.
+ *
+ * Currently, this provider does not have any configuration methods.
+ */
+angular.module('stormpath.socialLogin', ['stormpath.CONFIG'])
+.config(['$injector', 'STORMPATH_CONFIG', function $socialLoginProvider($injector, STORMPATH_CONFIG) {
+  /**
+   * @ngdoc object
+   *
+   * @name stormpath.socialLoginService.$socialLogin
+   *
+   * @description
+   *
+   * The social login service provides methods for letting users logging in with Facebook, Google, etc.
+   */
+  var socialLoginFactory = ['$http', '$q', '$injector', function socialLoginFactory($http, $q, $injector) {
+    var providers;
+
+    function SocialLoginService() {
+      return this;
+    }
+
+    /**
+     * @ngdoc function
+     *
+     * @name  stormpath.socialLoginService.$socialLogin#getProviders
+     *
+     * @methodOf stormpath.socialLoginService.$socialLogin
+     *
+     * @returns {promise}
+     *
+     * A promise that is resolved with the list of all available OAuth providers.
+     *
+     * @description
+     *
+     * Returns a list of all OAuth providers, provided by the `/oauth/providers` endpoint.
+     */
+    SocialLoginService.prototype.getProviders = function getProviders() {
+      var deferred = $q.defer();
+
+      if (providers) {
+        deferred.resolve(providers);
+        return deferred.promise;
+      }
+
+      $http.get(STORMPATH_CONFIG.getUrl('OAUTH_PROVIDERS_ENDPOINT')).then(function(response) {
+        providers = response.data;
+
+        Object.keys(providers).forEach(function(providerName) {
+          var provider = providers[providerName];
+          var service;
+
+          try {
+            service = $injector.get('$' + providerName + 'Login');
+          } catch (err) {
+            return;
```

Comment:
Ok. Though I think that `getProviders()` is doing a bit too much. What it actually does is not only *get providers*, it *initializes* them. So I'd rather have it as two separate methods. `initializeProviders()` that sets up all available providers and produces a list of initialized providers, and `getProviders()` that reads from that previous list.